src/Data/Swagger/Internal/Schema.hs

```diff
@@ -142,14 +142,14 @@ class ToSchema a where
   -- together with all used definitions.
   -- Note that the schema itself is included in definitions
   -- only if it is recursive (and thus needs its definition in scope).
-  declareNamedSchema :: proxy a -> Declare (Definitions Schema) NamedSchema
+  declareNamedSchema :: Proxy a -> Declare (Definitions Schema) NamedSchema
   default declareNamedSchema :: (Generic a, GToSchema (Rep a), TypeHasSimpleShape a "genericDeclareNamedSchemaUnrestricted") =>
-    proxy a -> Declare (Definitions Schema) NamedSchema
+    Proxy a -> Declare (Definitions Schema) NamedSchema
   declareNamedSchema = genericDeclareNamedSchema defaultSchemaOptions
 
 -- | Convert a type into a schema and declare all used schema definitions.
-declareSchema :: ToSchema a => proxy a -> Declare (Definitions Schema) Schema
-declareSchema = fmap _namedSchemaSchema . declareNamedSchema
+declareSchema :: forall a proxy. ToSchema a => proxy a -> Declare (Definitions Schema) Schema
```

Comment:
Or should this functions use concrete `Proxy` too? I'm not sure.

- The class member type-signature change is already a breaking change, so we can break more
- is there value in consistency?

I'm quite sure I used some of of `swagger2` combinators with not-`Proxy`, but it was indeed more of "cleverness" than strict requirement (i.e. *look*, I actually have `Maybe a` value here, I can plug).

Reply:
@fizruk opinions?

Reply:
That's a good point. 

Changing them all has the advantage that nobody will be confused about this difference, even if it otherwise doesn't matter in any significant way.

Reply:
I think it makes sense to switch to `Proxy` everywhere eventually. And since we are doing a breaking change, why not break all of it in the same way? 😊